src/composables/websocket.ts

```diff
@@ -0,0 +1,155 @@
+import { ref } from '@vue/composition-api';
+import camelcaseKeysDeep from 'camelcase-keys-deep';
+import { IDefaultComposableOptions, INetwork, WebSocketChannelTypes } from '../types';
+import { createNetworkWatcher } from './composablesHelpers';
+import { useGetter } from './vuex';
+
+const { onNetworkChange } = createNetworkWatcher();
+
+const wsClient = ref<WebSocket>();
+const isConnected = ref(false);
+const subscribers = ref<Record<WebSocketChannelTypes, Array<any>>>({
+  Transactions: [],
+  MicroBlocks: [],
+  KeyBlocks: [],
+});
+const channels = ref<Record<WebSocketChannelTypes, boolean>>({
+  Transactions: false,
+  MicroBlocks: false,
+  KeyBlocks: false,
+});
+
+export function useWebsocket({ store }: IDefaultComposableOptions) {
```

Comment:
As the native class has the "Socket" name uppercased I suggest to follow this naming: `useWebSocket`.